zetta_utils/convnet/architecture/convblock.py

```diff
@@ -44,9 +55,10 @@ def __init__(
         num_channels: List[int],
         activation: Callable[[], torch.nn.Module] = torch.nn.LeakyReLU,
         conv: Callable[..., torch.nn.modules.conv._ConvNd] = torch.nn.Conv2d,
-        padding_mode: Union[Literal["valid"], Literal["same"]] = "same",
         normalization: Optional[Callable[[int], torch.nn.Module]] = None,
         kernel_sizes: Union[List[int], int, List[Tuple[int, ...]], Tuple[int, ...]] = 3,
+        strides: Union[List[int], int, List[Tuple[int, ...]], Tuple[int, ...]] = 1,
+        paddings: Union[List[int], int, List[Tuple[int, ...]], Tuple[int, ...]] = 1,
```

Comment:
What's the usecase for paddings instead of the padding mode? Do people actually use paddings other than `same` and `valid`?

Reply:
The intention was that in might be useful in cases where the convs are transposed convolutions or strided

Reply:
Does `same` and `valid` do not what you want when using strided convolutions?

Fair point about transposed convs. I'm wondering if we should change the ConvBlock to only work for non-transposed convolutions, if it makes things simpler. Changing padding manually every time can be error prone.

Reply:
Strided conv don't work with the padding_mode I think:

"padding='valid' is the same as no padding. padding='same' pads the input so the output has the shape as the input. However, this mode doesn’t support any stride values other than 1."

Reply:
I see. How about we do the same thing as torch does -- accept either these integers or `same`, `valid` strings? Then we can default to `same` while letting an adventurous user to make it work for all kinds of convs.
The parameter should definitely be called `padding` like you did, not `padding_mode`

Reply:
Sounds good